external/external_custom/dockerfile/test.sh

```diff
@@ -0,0 +1,50 @@
+ #/bin/bash
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source $(dirname "$0")/test_base_functions.sh
+
+# Set up Java to be used by the external_custom test
+
+if [ -d /java/jre/bin ];then
+	echo "Using mounted Java8"
+	export JAVA_BIN=/java/jre/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+elif [ -d /java/bin ]; then
+	echo "Using mounted Java"
+	export JAVA_BIN=/java/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+else
+	echo "Using docker image default Java"
+	java_path=$(type -p java)
+	suffix="/java"
+	java_root=${java_path%$suffix}
+	export JAVA_BIN="$java_root"
+	$JAVA_BIN/java -version
+	export JAVA_HOME="${java_root%/bin}"
+fi
+
+echo_setup
+
+external_test_repo="$(echo ${EXTERNAL_CUSTOM_REPO} | awk -F'/' '{print $NF}' | sed 's/.git//g')"
+
+cd /external_test_repo
```

Comment:
Does this work without `$`?

Reply:
Yes the `make compile` works fine but `make _external_custom` does not work both with and without $. The whole output:
![output1](https://user-images.githubusercontent.com/59281636/124610131-daf24e80-de8d-11eb-9304-bfbb688f8acd.png)
![output2](https://user-images.githubusercontent.com/59281636/124610163-e2b1f300-de8d-11eb-9b45-c839394ac37c.png)
![output3](https://user-images.githubusercontent.com/59281636/124610192-ea719780-de8d-11eb-9ba1-f87b2a0591fb.png)
![output4](https://user-images.githubusercontent.com/59281636/124610220-f0677880-de8d-11eb-9eac-75d1d2918677.png)
![output7](https://user-images.githubusercontent.com/59281636/124610247-f6f5f000-de8d-11eb-878b-33520e43d2d3.png)
![output8](https://user-images.githubusercontent.com/59281636/124610278-ff4e2b00-de8d-11eb-9a75-44b984a08266.png)
![output9](https://user-images.githubusercontent.com/59281636/124610320-083efc80-de8e-11eb-97c7-b7f372b16824.png)

Reply:
The `test.sh` will run in the actual test execution (make _external_custom), not `make compile`. 
I will approve this PR, so we can run this in Grinder. Instead of debugging in your local, I think it will be easier to debug in Grinder.

Reply:
Thanks @llxia ! Could you please guide me a bit on how to run this in Grinder?

Reply:
By Grinder, I mean https://ci.adoptopenjdk.net/job/Grinder_Tweetest/. Once #2687 is merged, we can run tests on Jenkins.